webapp/js/components/AccountList.js

```diff
@@ -0,0 +1,195 @@
+            var message = "";
+            if (!data.success) {
+                message = data.message;
+            }
+            this.setState({keypairs: data.keypairs, message: message});
+        });
+    }
+
+    getModels() {
+        Models.list().then((response) => {
+            var data = JSON.parse(response.body);
+            var message = "";
+            if (!data.success) {
+                message = data.message;
+            }
+            this.setState({models: data.models, message: message});
+        });
+    }
+
+    // Indicates whether the key has everything uploaded for it
+    renderKeyStatus(acc) {
+        // Check if the key is used for signing system-users on any models
+        if (!this.state.models.find(m => (m['authority-id-user'] === acc.AuthorityID) & (m['key-id-user'] === acc.KeyID))) {
+            return <p>{T('not-used-signing')}</p>
+        }
+
+        // Check that we have an account assertion
+        var messages = []
+        if (!this.state.accounts.find(a => a.AuthorityID === acc.AuthorityID)) {
+            messages.push(T('no-assertion'))
+        }
+
+        // Check if we have an account key assertion
+        if ((!acc.Assertion) || (acc.Assertion.length === 0)) {
+            messages.push(T('no-assertion-key'))
+        }
+        
+        if (messages.length === 0) {
+            return (
+                <div>
+                    <pre className="code">{acc.Assertion}</pre>
+                </div>
+            )
+        }
+
+        return (
+            <div>
+                {messages.map((m, index, array) => {
+                    return (
+                        <p key={index}><i className="fa fa-exclamation-triangle warning"></i> {m}</p>
+                    )
+                })}
+            </div>
+        )
+    }
+
+    renderAccounts() {
+        if (this.state.accounts.length > 0) {
+            return (
+                <table>
+                <thead>
+                    <tr>
+                    <th className="small"></th><th>{T('authority-id')}</th><th>{T('assertion')}</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {this.state.accounts.map((acc) => {
+                    return (
+                        <tr key={acc.ID}>
+                            <td></td>
+                            <td>{acc.AuthorityID}</td>
+                            <td><pre className="code">{acc.Assertion}</pre></td>
+                        </tr>
+                    );
+                    })}
+                </tbody>
+                </table>
+            );
+        } else {
+            return (
+                <p>{T('no-assertions')}</p>
+            );
+        }
+    }
+
+    renderAccountKeys() {
+        if (this.state.keypairs.length > 0) {
+            return (
+                <table>
+                <thead>
+                    <tr>
+                    <th>{T('key-id')}</th><th>{T('status')}</th>
```

Comment:
Maybe having "assertion" as label instead of status would be better. In case all is ok, the assertion is shown as value for the status column, what looks a bit weird